src/Compilers/Core/Portable/MetadataReference/Metadata.cs

```diff
@@ -14,6 +14,11 @@ namespace Microsoft.CodeAnalysis
     /// </summary>
     public sealed class MetadataId
     {
+        private MetadataId()
+        {
+        }
+
+        internal static MetadataId CreateNewId() => new MetadataId();
```

Comment:
+1 to `Create`. That's the pattern in the compiler code base (e.g., `Compilation.Create`)